File: matrix-multiplication-speed/MATLAB/matrix_multiplication.m
LOOP_COUNT = 100;

m = 2000;
p = 200; 
n = 1000;

A = zeros(m, p);
B = zeros(p, n);

for i=1:m
    for j=1:p
        A(i,j) = i*p+j;
    end
end

for i=1:p
    for j=1:n
        B(i,j) = -(i*n+j);
    end
end

fprintf('Making the first run of matrix product using the * operator\n');
fprintf('to get stable run time measurements.\n\n');
C = A * B;

fprintf('Measuring performance of matrix product using the * operator.\n\n');
tic
for i=1:LOOP_COUNT
    C = A * B;
end
elapsed_time = toc / LOOP_COUNT;
fprintf('Elapsed time: %d milliseconds.\n', elapsed_time * 1000);

File: matrix-multiplication-speed/Python/matrix_multiplication.py
import time
import numpy as np

LOOP_COUNT = 100

m = 2000
p = 200
n = 1000

A = np.zeros((m, p))
B = np.zeros((p, n))

for i in np.arange(m):
    for j in np.arange(p):
        A[i, j] = i * p + j + 1
        
for i in np.arange(p):
    for j in np.arange(n):
        B[i, j] = -(i * n + j) - 1

print('Making the first run of matrix product  using numpy.matmul')
print('to get stable run time measurements.\n')
C = np.matmul(A, B)

print('Measuring performance of matrix product using numpy.matmul.\n')
start_time = time.time()
for i in np.arange(LOOP_COUNT):
    C = np.matmul(A, B)
elapsed_time = (time.time() - start_time) / LOOP_COUNT
print(f'Elapsed time: {elapsed_time * 1000:.3f} milliseconds.')
